### PR TITLE
WT-6416 RTS to skip traversing unnecessary internal pages

### PR DESCRIPTION
**Pull request by kommiharibabu**
Instead of traversing all the internal pages to find out the leaf
pages that have newer updates to be aborted, skip traversing some
of the internal pages when they are not present in the memory and
their aggregated durable timestamp is less than stable timestamp.

If an internal page is not in-memory tells that the entire sub tree
of that internal page must not be present in-memory.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -818,6 +818,40 @@ __rollback_abort_newer_updates(
     return (ret);
 }
 
+/*
+ * __wt_rts_page_skip --
+ *     Skip if rts doesn't requires to read this page.
+ */
+int
```
Comment (quchenhao) [inline src/txn/txn_rollback_to_stable.c:825]: Why not just return a bool in this function?

Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c:825]: The `__wt_rts_page_skip` function is passed as a function pointer to the tree walk function. This must follow same signature of the function pointer, so it cannot be changed.

File: src/txn/txn_rollback_to_stable.c [hunk 2]
```diff
@@ -818,6 +818,40 @@ __rollback_abort_newer_updates(
     return (ret);
 }
 
+/*
+ * __wt_rts_page_skip --
+ *     Skip if rts doesn't requires to read this page.
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c]: Fixed.

**Comment by agorrod** [comment]
@kommiharibabu I think this one is good to go. There is one more stress test pending, but that's a lot more test coverage than we normally have before merging.

**Pull request merged**